tools/rpkg/inst/include/cpp11/external_pointer.hpp

```diff
@@ -2,174 +2,178 @@
 // vendored on: 2022-01-10
 #pragma once
 
-#include <cstddef>      // for nullptr_t, NULL
-#include <memory>       // for bad_weak_ptr
-#include <type_traits>  // for add_lvalue_reference
+#include <cstddef>     // for nullptr_t, NULL
+#include <memory>      // for bad_weak_ptr
+#include <type_traits> // for add_lvalue_reference
 
-#include "cpp11/R.hpp"         // for SEXP, SEXPREC, TYPEOF, R_NilValue, R_C...
-#include "cpp11/protect.hpp"   // for protect, safe, protect::function
-#include "cpp11/r_bool.hpp"    // for r_bool
-#include "cpp11/r_vector.hpp"  // for type_error
-#include "cpp11/sexp.hpp"      // for sexp
+#include "cpp11/R.hpp"        // for SEXP, SEXPREC, TYPEOF, R_NilValue, R_C...
+#include "cpp11/protect.hpp"  // for protect, safe, protect::function
+#include "cpp11/r_bool.hpp"   // for r_bool
+#include "cpp11/r_vector.hpp" // for type_error
+#include "cpp11/sexp.hpp"     // for sexp
 
 namespace cpp11 {
 
 template <typename T>
-void default_deleter(T* obj) {
-  delete obj;
+void default_deleter(T *obj) {
+	delete obj;
 }
 
-template <typename T, void Deleter(T*) = default_deleter<T>>
+template <typename T, void Deleter(T *) = default_deleter<T>>
 class external_pointer {
- private:
-  sexp data_ = R_NilValue;
-
-  static SEXP valid_type(SEXP data) {
-    if (data == nullptr) {
-      throw type_error(EXTPTRSXP, NILSXP);
-    }
-    if (TYPEOF(data) != EXTPTRSXP) {
-      throw type_error(EXTPTRSXP, TYPEOF(data));
-    }
-
-    return data;
-  }
-
-  static void r_deleter(SEXP p) {
-    if (TYPEOF(p) != EXTPTRSXP) return;
-
-    T* ptr = static_cast<T*>(R_ExternalPtrAddr(p));
-
-    if (ptr == NULL) {
```

Comment:
I don't see any vector-specific change here, so I would avoid touching this file.